azurelinuxagent/common/logcollector.py

```diff
@@ -0,0 +1,340 @@
+# Microsoft Azure Linux Agent
+#
+# Copyright 2020 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+
+import glob
+from heapq import heappush, heappop
+import logging
+import os
+import subprocess
+import zipfile
+
+from azurelinuxagent.common.conf import get_lib_dir, get_ext_log_dir
+from azurelinuxagent.common.future import ustr
+from azurelinuxagent.common.utils.fileutil import mkdir, read_file, write_file, rm_files
```

Comment:
We need to be very careful adding dependencies to other modules in the agent, since the agent uses several globals. For example, if at some point we add a dependency on a function that uses logger then that output would be in the agent's log and may be confusing. Telemetry is also global. Your current dependencies look pretty safe, but if it was me I would get rid of the dependency on fileutil since I do not think those functions are buying you much.

Also, please add this note as a comment so that people changing the code do not add undesired dependencies.

Reply:
Thanks, this is a very good point that hadn't crossed my mind. I agree that the fileutil dependencies are not worth keeping, I will refactor them.